src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs

```diff
@@ -427,6 +429,90 @@ public async Task SslStream_ClientCertificate_SendsChain()
             }
         }
 
+        [Theory]
+        [InlineData(16384 * 100, 4096, 1024, true)]
+        [InlineData(16384 * 100, 1024 * 20, 1024, true)]
+        public async Task SslStream_RandomWrites_OK(int bufferSize, int readBufferSize, int writeBufferSize, bool useAsync)
+        {
+            byte[] dataToCopy = RandomNumberGenerator.GetBytes(bufferSize);
+            byte[] srcHash = SHA256.Create().ComputeHash(dataToCopy);
+
+            var clientOptions = new SslClientAuthenticationOptions() { TargetHost = "localhost", };
+            clientOptions.RemoteCertificateValidationCallback = (sender, certificate, chain, sslPolicyErrors) => true;
+
+            var serverOptions = new SslServerAuthenticationOptions();
+            serverOptions.ServerCertificateContext = SslStreamCertificateContext.Create(Configuration.Certificates.GetServerCertificate(), null);
+
+            (Stream clientStream, Stream serverStream) = TestHelper.GetConnectedTcpStreams();
+            using (clientStream)
+            using (serverStream)
+            using (SslStream client = new SslStream(new RandomIOStream(clientStream, readBufferSize)))
+            using (SslStream server = new SslStream(new RandomIOStream(serverStream)))
+            {
+                Task t1 = client.AuthenticateAsClientAsync(clientOptions, CancellationToken.None);
+                Task t2 = server.AuthenticateAsServerAsync(serverOptions, CancellationToken.None);
+
+                await TestConfiguration.WhenAllOrAnyFailedWithTimeout(t1, t2);
+
+                Task writer = Task.Run(() =>
+                {
+                    Memory<byte> data = new Memory<byte>(dataToCopy);
+                    while (data.Length > 0)
+                    {
+                        int writeLength = Math.Min(data.Length, writeBufferSize);
+                        if (useAsync)
+                        {
+                            server.WriteAsync(data.Slice(0, writeLength)).GetAwaiter().GetResult();
+                        }
+                        else
+                        {
+                            server.Write(data.Span.Slice(0, writeLength));
+                        }
+
+                        data = data.Slice(writeBufferSize);
+                    }
+
+                    server.ShutdownAsync().GetAwaiter().GetResult();
+                });
+
+                Task reader = Task.Run(() =>
+                {
+                    SHA256 hash = SHA256.Create();
```

Comment:
```suggestion
                    using SHA256 hash = SHA256.Create();
```

Reply:
I thought we had static's now

Reply:
Ah yes, you can just do `SHA256.HashData(...)`. No Create() and no dispose.

Reply:
That works only when you have whole chunk. Here we feed `TransformFinalBlock` one chunk at the time.